src/Microsoft.ML.DataView/VBufferEditor.cs

```diff
@@ -139,15 +148,16 @@ public VBuffer<T> Commit()
         /// The newly created <see cref="VBuffer{T}"/>.
         /// </returns>
         /// <remarks>
-        /// CommitTruncated allows to modify the length of the explicitly
-        /// defined values.
+        /// This method allows to modify the length of the explicitly defined values.
         /// This is useful in sparse situations where the <see cref="VBufferEditor{T}"/>
         /// was created with a larger physical value count than was needed
         /// because the final value count was not known at creation time.
         /// </remarks>
         public VBuffer<T> CommitTruncated(int physicalValuesCount)
         {
-            Contracts.CheckParam(physicalValuesCount <= Values.Length, nameof(physicalValuesCount), "Updating physicalValuesCount during CommitTruncated cannot be greater than the original physicalValuesCount value used in Create.");
+            Contracts.CheckParam(physicalValuesCount <= Values.Length, nameof(physicalValuesCount),
+                "Updating " + nameof(physicalValuesCount) + " during " + nameof(CommitTruncated) +
```

Comment:
>"Updating "  [](start = 16, length = 12)

nit: perhaps convert it to a more modern version of
$"Updating {nameof(physicalValuesCount) } during { nameof(CommitTruncated)}"+ ...

Reply:
Hi @glebuk, just a quick note, this advice is incorrect. Try the following for yourself (relies on there being a name called values.

```csharp
const string a = "Hello " + nameof(values);
const string b = $"Hello {nameof(values)}";
```

You will note that the first will succeed, but the second will fail because this is not actually a constant. The reason this is relevant is, you'll remember I hope our advice that the strings in checks or asserts and whatnot should always be *constants*, so as to avoid needless composition of error messages that are never used.

Happily, even if i had committed this error, I actually wrote a code analyzer to stop people from making such a mistake, but it might be good to remember why this is a bad idea.

Reply:
Thanks!

---
In reply to: [270520617](https://github.com/dotnet/machinelearning/pull/3136#discussion_r270520617) [](ancestors = 270520617)

Reply:
Just a side note to this conversation (it won't directly affect this scenario),

The .NET team is working on making these string formatting more efficient. Check out https://github.com/dotnet/csharplang/issues/2302 if you are interested in learning more.